esmvaltool/cmorizers/data/formatters/datasets/esacci_soilmoisture.py

```diff
@@ -0,0 +1,165 @@
+
+
+    Returns
+    -------
+    cube: iris.cube.Cube
+        data cube with fixed coordinates.
+    """
+    # first fix any completely missing coord var names
+    fix_dim_coordnames(cube)
+    # fix individual coords
+    for cube_coord in cube.coords():
+        # fix time
+        if cube_coord.var_name == 'time':
+            logger.info("Fixing time...")
+            cube.coord('time').convert_units(
+                Unit('days since 1970-01-01T00:00:00+00:00',
+                     calendar='proleptic_gregorian'))
+
+        # fix longitude
+        if cube_coord.var_name == 'lon':
+            logger.info("Fixing longitude...")
+            if cube_coord.ndim == 1:
+                if cube_coord.points[0] < 0. and \
+                        cube_coord.points[-1] < 181.:
+                    cube_coord.points = \
+                        cube_coord.points + 180.
+                    cube.attributes['geospatial_lon_min'] = 0.
+                    cube.attributes['geospatial_lon_max'] = 360.
+                    nlon = len(cube_coord.points)
+                    roll_cube_data(cube, nlon // 2, -1)
+        fix_bounds(cube, cube_coord)
+
+    return cube
+
+
+def extract_variable(var_info, raw_info, year):
+    """Extract variables."""
+    rawvar = raw_info['name']
+    constraint = iris.Constraint(name=rawvar)
+    if rawvar == 'sm_uncertainty':
+        sm_cube = iris.load_cube(raw_info['file'],
+                                 iris.NameConstraint(var_name='sm'))
+        ancillary_var = sm_cube.ancillary_variable(
+            'Volumetric Soil Moisture Uncertainty'
+        )
+        cube = sm_cube.copy(ancillary_var.core_data())
+    else:
+        cube = iris.load_cube(raw_info['file'], constraint)
+
+    # Continue with processing the cube if it was successfully loaded
+    fix_var_metadata(cube, var_info)
+    convert_timeunits(cube, year)
+    fix_coords(cube)
+
+    # Remove dysfunctional ancillary data without standard names
+    for ancillary_variable in cube.ancillary_variables():
+        cube.remove_ancillary_variable(ancillary_variable)
+
+    return cube
+
+
+def cmorization(in_dir, out_dir, cfg, cfg_user, start_date, end_date):
+    """Cmorize data."""
+    glob_attrs = cfg['attributes']
+
+    # run the cmorization
+    for var_name, vals in cfg['variables'].items():
+        all_data_cubes = []
+        if not isinstance(vals, dict):  # Ensure vals is a dictionary
+            raise ValueError(
+                f"Invalid format for variable {var_name}: {type(vals)}"
+            )
+        var_info = cfg['cmor_table'].get_variable(vals['mip'], var_name)
+        glob_attrs['mip'] = vals['mip']
+        raw_info = {'name': vals['raw']}
+        inpfile_pattern = os.path.join(in_dir, vals['filename'])
+        logger.info("CMORizing var %s from file type %s",
+                    var_name, inpfile_pattern)
+
+        for year in range(vals['start_year'], vals['end_year'] + 1):
```

Comment:
```suggestion
        for year in range(start_date.year, end_date.year + 1):
```

It would be nice to allow the user to make use of the `--start` and `--end` CLI arguments. You can also get rid of the `start_year` and `end_year` options in the config file.

Reply:
Thanks, I have removed the start_year and end_year options in the config file and the user can now specify them by using the --start and --end CLI arguments indeed. If they are missing the start_date and end_date are taken from above.